client/httplib/httpclient.go

```diff
@@ -0,0 +1,296 @@
+	return c
+}
+
+// HeaderValue will set response HeaderValue to the pointer
+func (c *Client) HeaderValue(key string, value **string) *Client {
+	if c.pointer == nil {
+		newC := *c
+		newC.pointer = &responsePointer{
+			headerValues: map[string]**string{
+				key: value,
+			},
+		}
+		return &newC
+	}
+	if c.pointer.headerValues == nil {
+		c.pointer.headerValues = map[string]**string{}
+	}
+	c.pointer.headerValues[key] = value
+	return c
+}
+
+// ContentType will set response ContentType to the pointer
+func (c *Client) ContentType(contentType **string) *Client {
+	return c.HeaderValue("Content-Type", contentType)
+}
+
+// ContentLength will set response ContentLength to the pointer
+func (c *Client) ContentLength(contentLength **int64) *Client {
+	if c.pointer == nil {
+		newC := *c
+		newC.pointer = &responsePointer{
+			contentLength: contentLength,
+		}
+		return &newC
+	}
+	c.pointer.contentLength = contentLength
+	return c
+}
+
+// setPointers set the http response value to pointer
+func (c *Client) setPointers(resp *http.Response) {
+	if c.pointer == nil {
+		return
+	}
+	if c.pointer.response != nil {
+		*c.pointer.response = resp
+	}
+	if c.pointer.statusCode != nil {
+		*c.pointer.statusCode = &resp.StatusCode
+	}
+	if c.pointer.header != nil {
+		*c.pointer.header = &resp.Header
+	}
+	if c.pointer.headerValues != nil {
+		for k, v := range c.pointer.headerValues {
+			s := resp.Header.Get(k)
+			*v = &s
+		}
+	}
+	if c.pointer.contentLength != nil {
+		*c.pointer.contentLength = &resp.ContentLength
+	}
+}
+
+// initRequest will apply all the client setting, common option and request option
+func (c *Client) newRequest(method, path string, opts []BeegoHttpRequestOption) (*BeegoHTTPRequest, error) {
+	var req *BeegoHTTPRequest
+	switch method {
+	case http.MethodGet:
+		req = Get(c.Endpoint + path)
+	case http.MethodPost:
+		req = Post(c.Endpoint + path)
+	case http.MethodPut:
+		req = Put(c.Endpoint + path)
+	case http.MethodDelete:
+		req = Delete(c.Endpoint + path)
+	case http.MethodHead:
+		req = Head(c.Endpoint + path)
+	}
+
+	req = req.Setting(*c.Setting)
+	for _, o := range c.CommonOpts {
+		err := o(req)
+		if err != nil {
+			return nil, err
+		}
+	}
+	for _, o := range opts {
+		err := o(req)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return req, nil
+}
+
+// handleResponse try to parse body to meaningful value
+func (c *Client) handleResponse(value interface{}, req *BeegoHTTPRequest) error {
+	// send request
+	resp, err := req.Response()
+	if err != nil {
+		return err
+	}
+	c.setPointers(resp)
+
+	if value == nil {
+		return nil
+	}
+
+	// handle basic type
+	switch v := value.(type) {
+	case **string:
+		s, err := req.String()
+		if err != nil {
+			return nil
+		}
+		*v = &s
+		return nil
+	case **[]byte:
+		bs, err := req.Bytes()
+		if err != nil {
+			return nil
+		}
+		*v = &bs
+		return nil
+	}
+
+	// try to parse it as content type
+	switch strings.Split(resp.Header.Get("Content-Type"), ";")[0] {
```

Comment:
Thinking about the case that we don't know the `Content-Type` which means that server didn't return this header.

Reply:
Here I do three scenarios of return. 

One is that when the value is **string or **byte, the content will be directly put into value as string or [] byte and returned. 

The second is to try to judge the type by the value of its header, and parse and return. 

The third is to try to use various types of parsing directly when contenttype does not exist, and then put back the result that the parsing will not report an error.